include/rcpputils/filesystem_helper.hpp

```diff
@@ -122,6 +130,14 @@ class path
     return path_.empty() ? path() : *--this->cend();
   }
 
+  std::string extension() const
```

Comment:
This class is a drop-in replacement of the `std::filesystem::path` class. So we should match its interface: https://en.cppreference.com/w/cpp/filesystem/path/extension

If you really want a string, you could always make that a free funtion to avoid the problem.

Reply:
+1, it should match the `std::filesystem` API. The `path` type has a `.string()` and `.c_str()`.

Reply:
I updated it, it now returns empty string for no extension, and a path object for the extension. Let me know if it looks ok, I updated the tests as well.